app/uk/gov/hmrc/bindingtariffrulingfrontend/views/components/input_search.scala.html

```diff
@@ -63,9 +63,17 @@
   ))
 
   @govukButton(Button(
-    attributes = Map("id" -> "submit"),
+    attributes = Map(
+        "id" -> "submit",
+        "onclick" -> "window.dataLayer.push({ 'search_query_atar_ruling': searchParam })"),
     inputType = Some("submit"),
     content = Text(messages("search.form.submit"))
   ))
+
+    <script>
+        const urlParams = new URLSearchParams(window.location.search);
+        const searchParam = urlParams.get('query');
+        if (searchParam) window.dataLayer.push({'search_query_atar_ruling': searchParam});
```

Comment:
I think this code here is going to add a data layer entry regardless of whether the user clicks the button - should it be removed? You will most likely still need the two variables to be declared before the button though